From 9e503d7068889e52e9144079de331ed51676e535 Mon Sep 17 00:00:00 2001
From: Chris Dent <chdent@redhat.com>
Date: Tue, 13 Oct 2015 14:27:09 +0000
Subject: [PATCH] If mongodb_replica_set is empty do not pass it to client

Otherwise a replicaSet parameter in a db connection URL will be
clobbered by the empty configuration setting.

Note that this is a direct patch to stable/liberty, not a backport from
master because master has changed such that there is no longer a
mongodb_replica_set configuration setting.

Change-Id: Ic8de93f58ff1cf21888719597490f987d703cab6
Closes-Bug: #1505669
Upstream-Liberty: https://review.openstack.org/234254
(cherry picked from commit d91e1916c6c34892dd1026954bc7327ee4c1898e)
---
 ceilometer/storage/mongo/utils.py | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/ceilometer/storage/mongo/utils.py b/ceilometer/storage/mongo/utils.py
index 8cc588b..e2a2650 100644
--- a/ceilometer/storage/mongo/utils.py
+++ b/ceilometer/storage/mongo/utils.py
@@ -269,11 +269,14 @@ class ConnectionPool(object):
     @staticmethod
     def _mongo_connect(url):
         try:
-            client = MongoProxy(
-                pymongo.MongoClient(
-                    url, replicaSet=cfg.CONF.database.mongodb_replica_set
+            if cfg.CONF.database.mongodb_replica_set:
+                client = MongoProxy(
+                    pymongo.MongoClient(
+                        url, replicaSet=cfg.CONF.database.mongodb_replica_set
+                    )
                 )
-            )
+            else:
+                client = MongoProxy(pymongo.MongoClient(url))
             return client
         except pymongo.errors.ConnectionFailure as e:
             LOG.warn(_('Unable to connect to the database server: '
